### PR TITLE
Added RMS normalization layer

### LINKED ISSUE #2849: Incorporate RMSNorm
Is there interest to incorporate RMSNorm (https://arxiv.org/pdf/1910.07467.pdf) either as a seperate normalization module or to make LayerNorm configurable to switch off the centering? It is becoming quite popular as a faster alternative to LayerNorm that maintains performance and is used for example in Palm.

### PR DESCRIPTION
**Pull request by chiamp**
Resolves #2849.

Added an optional argument `use_mean` in the `_compute_stats` function in `flax/linen/normalization.py`, which will compute the mean and variance if set to `True`, and will set the mean to 0 and compute the variance without subtracting the mean if set to `False`. The latter mode is useful as square rooting this "variance" value (which is done in the `_normalize` function) will give you the RMS.

### CODE REVIEW

File: flax/linen/normalization.py [hunk 1]
```diff
@@ -335,6 +343,70 @@ def __call__(self, x):
         self.bias_init, self.scale_init)
 
 
+class RMSNorm(Module):
```
Comment (marcvanzee) [inline flax/linen/normalization.py:346]: Could you please add an example of how to use the layer? I think we should starting doing this for every layer, like @cgarciae does in his RNN PR: https://github.com/google/flax/pull/2604/files#r1107264719.

**Review by levskaya — APPROVED** [review]
Looks good!  Thanks!! -- I second Marc's ask to add a small usage example in the docstring.

**Comment by chiamp** [comment]
@marcvanzee @levskaya I added a [docstring](https://github.com/google/flax/pull/2881/files#diff-f025750faae1b20976622f833c327e7c3c9bf25f48a5de676eb9744febd0d8a5R356), let me know if this works!

**Review by levskaya — APPROVED** [review]

**Comment by levskaya** [comment]
@chiamp - I added a exception for the deprecation warning, your tests all seem to pass now!

**Comment by chiamp** [comment]
> @chiamp - I added a exception for the deprecation warning, your tests all seem to pass now!

Thanks @levskaya!

File: flax/linen/normalization.py [hunk 2]
```diff
@@ -335,6 +343,80 @@ def __call__(self, x):
...
+    axis_index_groups: groups of axis indices within that named axis
+      representing subsets of devices to reduce over (default: None). For
+      example, `[[0, 1], [2, 3]]` would independently batch-normalize over
+      the examples on the first two and last two devices. See `jax.lax.psum`
+      for more details.
+  """
+  epsilon: float = 1e-6
+  dtype: Optional[Dtype] = None
+  param_dtype: Dtype = jnp.float32
+  use_bias: bool = True
```
Comment (levskaya) [inline flax/linen/normalization.py:390]: sorry I just noticed this - we probably don't want `use_bias` and `bias_init` here since we're never adjusting the offset?
